research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,306 @@
+                    'Sigla Partido': 'party_acronym',
+                    'Sigla UE doador': 'donor_electoral_unit_abbreviation',
+                    'Tipo de documento': 'document_type',
+                    'Tipo doador originário': 'original_donor_type',
+                    'Tipo partido': 'party_type',
+                    'Tipo receita': 'revenue_type',
+                    'Tipo comite': 'committee_type',
+                    'UF': 'state',
+                    'Valor receita': 'revenue_value'}
+    if 'candidates' in donations_data.keys():
+        donations_data['candidates'].rename(columns={translations},
+                                            inplace=True)
+    if 'parties' in donations_data.keys():
+        donations_data['parties'].rename(columns={translations},
+                                         inplace=True)
+    if 'committees' in donations_data.keys():
+        donations_data['committees'].rename(columns={translations},
+                                            inplace=True)
+
+
+def download_base(url, folder_walk_function):
+    file_name = url.split('/')[-1]
+    print("Downloading " + file_name)
+    urllib.request.urlretrieve(url, file_name, reporthook)
+    print("Uncompressing downloaded data.")
+    with zipfile.ZipFile(file_name, "r") as zip_ref:
+        zip_ref.extractall(file_name.split('.')[0])
+    os.remove(file_name)
+
+    print("Reading all the data and creating the dataframes...")
+
+    donations_data = folder_walk_function()
+
+    if 'candidates' in donations_data.keys():
+        columns_candidates = donations_data['candidates'].columns.values
+        columns_mod_candidates = {}
+        for name in columns_candidates:
+            columns_mod_candidates[name] = name.strip()
+        donations_data['candidates'].rename(columns=columns_mod_candidates,
+                                            inplace=True)
+
+    if 'parties' in donations_data.keys():
+        columns_parties = donations_data['parties'].columns.values
+        columns_mod_parties = {}
+        for name in columns_parties:
+            columns_mod_parties[name] = name.strip()
+        donations_data['parties'].rename(columns=columns_mod_parties,
+                                         inplace=True)
+
+    if 'committees' in donations_data.keys():
+        columns_committees = donations_data['committees'].columns.values
+        columns_mod_committees = {}
+        for name in columns_committees:
+            columns_mod_committees[name] = name.strip()
+        donations_data['committees'].rename(columns=columns_mod_committees,
+                                            inplace=True)
+
+    return donations_data
+
+if __name__ == "__main__":
+    base_url = ('http://agencia.tse.jus.br/estatistica/sead/odsele/'
+                'prestacao_contas/')
+    url = base_url + 'prestacao_contas_2010.zip'
+    donations_data_2010 = download_base(url, folder_walk_2010)
```

Comment:
In here too, we don't need to have all those lines that may only change the year, we can use one and then change the year as it is needed :)